pallets/treasury/src/lib.rs

```diff
@@ -371,7 +383,7 @@ pub mod pallet {
 
             // allocate to beneficiary
             T::Currency::deposit_into_existing(&who, BalanceOf::<T>::from(total_rewards))
-                .expect("Should deposit balance to account"); //TODO
+                .expect("Should deposit balance to account"); // TODO
 
```

Comment:
i believe
+ we should introduce the factory https://docs.rs/pallet-treasury/3.0.0/pallet_treasury/
+ refactor the inflation mechanism in the existing pallet-treasury stub back to sth like pallet-inflation
+ while keeping things in line with the new circuit timing pallet